src/js/vic-v2/components/PhotoField.jsx

```diff
@@ -231,31 +267,73 @@ export default class PhotoField extends React.Component {
     this.setState({ dragAndDropSupported: dragAndDropSupported && !iOS });
   }
 
-  detectWidth = () => {
-    const windowWidth = window.innerWidth;
-    this.setState({ windowWidth });
+  updateBoundaryWarningAndButtonStates = () => {
+    const canvasCropBoundariesMet = this.getCanvasCropBoundariesMet(this.refs.cropper.getCanvasData(), this.refs.cropper.getCropBoxData());
+
+    this.updateMoveButtonsEnabledStates(canvasCropBoundariesMet);
+    this.maybeShowCropBoundaryWarning(canvasCropBoundariesMet);
+  }
+
+  updateMoveButtonsEnabledStates = ({ topBoundaryMet, bottomBoundaryMet, rightBoundaryMet, leftBoundaryMet }) => {
+    const buttonStates = {
+      moveUpDisabled: bottomBoundaryMet,
+      moveDownDisabled: topBoundaryMet,
+      moveLeftDisabled: rightBoundaryMet,
+      moveRightDisabled: leftBoundaryMet
+    };
+
+    this.setState(buttonStates);
+  }
+
+  makeBoundaryEdgeWarning = (direction) => `You have reached the edge of your photo and cannot move it any farther ${direction}. Use the other arrows or the make larger and make smaller buttons to continue to edit.`
+
+  maybeShowCropBoundaryWarning = ({ zoomWarn = false, topBoundaryMet, bottomBoundaryMet, rightBoundaryMet, leftBoundaryMet }) => {
+    let warningMessage;
+    if (zoomWarn) {
+      warningMessage = 'If you zoom in this close, your ID photo will be less clear';
+    } else if ((topBoundaryMet && bottomBoundaryMet) || (leftBoundaryMet && rightBoundaryMet)) {
+      warningMessage = 'Your photo currently fits within the square frame. Click ‘Make Larger’ if you would like to adjust the position of your photo';
+    } else if (topBoundaryMet) {
+      warningMessage = this.makeBoundaryEdgeWarning('down');
+    } else if (bottomBoundaryMet) {
+      warningMessage = this.makeBoundaryEdgeWarning('up');
+    } else if (leftBoundaryMet) {
+      warningMessage = this.makeBoundaryEdgeWarning('right');
+    } else if (rightBoundaryMet) {
+      warningMessage = this.makeBoundaryEdgeWarning('left');
+    } else {
+      warningMessage = '';
+    }
+
+    this.setState({ warningMessage });
+  }
+
+  cropend = () => { // casing matches Cropper argument
+    this.updateBoundaryWarningAndButtonStates();
   }
 
   moveUp = () => {
     this.refs.cropper.move(0, -5);
+
+    this.updateBoundaryWarningAndButtonStates();
   }
 
   moveDown = () => {
     this.refs.cropper.move(0, 5);
+
+    this.updateBoundaryWarningAndButtonStates();
   }
 
   moveRight = () => {
-    this.refs.cropper.move(-5, 0);
+    this.refs.cropper.move(5, 0);
```

Comment:
Not sure if this was intentional. Clicking `moveRight` moved the image left. I vaguely remember a conversation about this being desirable. LMK @cehsu